[BUG] build failure with -Wthread-safety-analysis, regression of d0fbf8ac238d3bc3a111334ab3553173812d78e7

**Describe the bug**
-Wthread-safety-analysis error found by compilation

**System**
OS: Ubuntu 20.04.3 LTS
Compiler: clang version 13.0.0 (https://github.com/llvm/llvm-project.git d7b669b3a30345cfcdb2fde2af6f48aa4b94845d)

**To reproduce**
source: (current master, in my case 28e1da21d8d677bc98f12ccc7fc159ff19e8e817)
```
$ /usr/bin/clang++ -Iinclude -Werror -Wthread-safety -DHAVE_THREAD_SAFETY_ATTRIBUTES -c -O2 src/benchmark.cc
In file included from src/benchmark.cc:18:
In file included from src/benchmark_runner.h:23:
src/perf_counters.h:139:33: error: reading variable 'counters_' requires holding mutex 'mutex_' [-Werror,-Wthread-safety-analysis]
  bool IsValid() const { return counters_.IsValid(); }
                                ^
src/perf_counters.h:146:20: error: reading variable 'counters_' requires holding mutex 'mutex_' [-Werror,-Wthread-safety-analysis]
    valid_read_ &= counters_.Snapshot(&start_values_);
                   ^
src/perf_counters.h:156:20: error: reading variable 'counters_' requires holding mutex 'mutex_' [-Werror,-Wthread-safety-analysis]
    valid_read_ &= counters_.Snapshot(&end_values_);
                   ^
src/perf_counters.h:159:28: error: reading variable 'counters_' requires holding mutex 'mutex_' [-Werror,-Wthread-safety-analysis]
    for (size_t i = 0; i < counters_.names().size(); ++i) {
                           ^
src/perf_counters.h:162:31: error: reading variable 'counters_' requires holding mutex 'mutex_' [-Werror,-Wthread-safety-analysis]
      measurements.push_back({counters_.names()[i], measurement});
                              ^
5 errors generated.
```

**Expected behavior**
no build error (worked before d0fbf8ac238d3bc3a111334ab3553173812d78e7)


I'm not sure if adding the mutex has unwanted performance implications. Before, there was no mutex at all, so I'm not sure what the right way to go forward would be. Maybe you have an idea, @taoliq ?

--- a/src/perf_counters.h
+++ b/src/perf_counters.h
@@ -136,10 +136,14 @@ class PerfCountersMeasurement final {
   // concurrently running with this. This is preferring efficiency to
   // maintainability, because the address of the static can be known at compile
   // time.
-  bool IsValid() const { return counters_.IsValid(); }
+  bool IsValid() const {
+    MutexLock l(mutex_);
+    return counters_.IsValid();
+  }
 
   BENCHMARK_ALWAYS_INLINE void Start() {
     assert(IsValid());
+    MutexLock l(mutex_);
     // Tell the compiler to not move instructions above/below where we take
     // the snapshot.
     ClobberMemory();
@@ -150,6 +154,7 @@ class PerfCountersMeasurement final {
   BENCHMARK_ALWAYS_INLINE bool Stop(
       std::vector<std::pair<std::string, double>>& measurements) {
     assert(IsValid());
+    MutexLock l(mutex_);
     // Tell the compiler to not move instructions above/below where we take
     // the snapshot.
     ClobberMemory();

